src/main/scala/ch/epfl/bluebrain/nexus/kg/resources/Repo.scala

```diff
@@ -91,31 +96,51 @@ class Repo[F[_]: Monad](agg: Agg[F], clock: Clock, toIdentifier: ResId => String
     * @param rev    the last known revision of the resource
     * @param attach the attachment description metadata
     * @param source the source of the attachment
+    * @param instant  an optionally provided operation instant
     * @tparam In the storage input type
     * @return either a rejection or the new resource representation in the F context
     */
-  def attach[In](id: ResId, rev: Long, attach: BinaryDescription, source: In)(
+  def attach[In](id: ResId, rev: Long, attach: BinaryDescription, source: In, instant: Instant = clock.instant)(
       implicit identity: Identity,
       store: AttachmentStore[F, In, _]): EitherT[F, Rejection, Resource] =
     get(id).toRight(NotFound(id.ref)).flatMap { res =>
       if (res.deprecated) EitherT.leftT(IsDeprecated(id.ref))
       else if (res.rev != rev) EitherT.leftT(IncorrectRev(id.ref, rev))
       else
         store.save(id, attach, source).flatMap { attr =>
-          evaluate(id, AddAttachment(id, rev, attr, clock.instant(), identity))
+          evaluate(id, AddAttachment(id, rev, attr, instant, identity))
         }
     }
 
+  /**
```

Comment:
can you add a deprecated on the docs here? We should remember to delete this as soon as we don't need v0 migrations and this is not on a v0 package so we might forget